### PR TITLE
*: add rollback for load data statement

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Add transaction rollback for load data.

### Check List 

Tests 

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @jackysp PTAL

File: server/conn.go [hunk 1]
```diff
@@ -954,19 +954,24 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
...
+			if err != nil {
+				if err1 := txn.Rollback(); err1 != nil {
+					log.Errorf("load data rollback failed: %v", err1)
+				}
+				return errors.Trace(err)
 			}
 		}
-		return errors.Trace(err)
+	} else {
+		log.Error(err1)
```
Comment (zimulala) [inline server/conn.go]: Please add some information like line967.
And If `err1 != nil` is true, I think we can't commit this transaction.

**Comment by tiancaiamao** [comment]
Please address comment, rest looks good to me

File: server/conn.go (hunk 1 above)
Comment (jackysp) [reply server/conn.go]: Maybe panic here will be better.

Comment (crazycs520) [reply server/conn.go]: done.

**Comment by crazycs520** [comment]
@tiancaiamao PTAL

**Pull request merged**